File: backend-dotnet/Application/Queries/Users/GetCurrentUserQueryHandler.cs
using Application.Interfaces;

namespace Application.Queries.Users
{
    public class GetCurrentUserQueryHandler : IQueryHandler<GetCurrentUserQuery, object>
    {
        private readonly IUserRepository _userRepository;

        public GetCurrentUserQueryHandler(IUserRepository userRepository)
        {
            _userRepository = userRepository;
        }

        public async Task<object> HandleAsync(GetCurrentUserQuery query)
        {
            var user = await _userRepository.GetByIdAsync(query.UserId);

            if (user == null)
                return null!;

            return new
            {
                user.Id,
                user.Username,
                user.Email,
                user.ProfileImageUrl,
                user.CreatedAt
            };
        }
    }
}

File: backend-java/src/main/java/com/jerrygram/application/queries/users/GetCurrentUserQueryHandler.java
package com.jerrygram.application.queries.users;

import com.jerrygram.application.dtos.UserProfileDto;
import com.jerrygram.application.interfaces.IQueryHandler;
import com.jerrygram.infrastructure.repositories.UserRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class GetCurrentUserQueryHandler implements IQueryHandler<GetCurrentUserQuery, UserProfileDto> {

    private final UserRepository userRepository;

    @Override
    public UserProfileDto handle(GetCurrentUserQuery query) {
        var userId = query.getUserId();

        var user = userRepository.findById(userId).orElse(null);

        if (user == null) {
            return null;
        }

        return UserProfileDto.builder()
                .id(user.getId())
                .username(user.getUsername())
                .email(user.getEmail())
                .profileImageUrl(user.getProfileImageUrl())
                .createdAt(user.getCreatedAt())
                .followers(user.getFollowersCount())
                .followings(user.getFollowingCount())
                .build();
    }
}